### PR TITLE
ResDiff Integration: Config Management

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/experimental/resdiff/conf/config_generate.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  # Sampling resolution
+pretext: null
+  # String to include the task
+res_edm: true
+  # If residual-based edm is used
+sampling_method: "stochastic"
+  # Sampling method ["stochastic", "deterministic"]
+
+## Data options
+train_data_path: "s3://sw_climate_fno/nbrenowitz/2023-01-24-cwb-4years.zarr"
```
Comment (loliverhennigh) [inline modulus/experimental/resdiff/conf/config_generate.yaml:43]: Isn't this path the internal one we have? If possible probably not include this.

File: modulus/experimental/resdiff/training/img_utils.py [hunk 2]
```diff
@@ -24,72 +24,77 @@ def reshape_fields(
...
     if normalize and train:
-        if params.normalization == "minmax":
+        if normalization == "minmax":
             img -= mins
             img /= maxs - mins
-        elif params.normalization == "zscore":
-            # print('params.normalization == zscore')
+        elif normalization == "zscore":
             img -= means
             img /= stds
```
Comment (Alexey-Kamenev) [inline modulus/experimental/resdiff/training/img_utils.py:78]: Not sure if you want to fix it in this PR, but `else` clause is missing here.

File: modulus/experimental/resdiff/training/img_utils.py [hunk 3]
```diff
@@ -101,12 +106,9 @@ def reshape_fields(
...
             img = np.concatenate((img, grid), axis=1)
 
-    if params.roll:
+    if roll:
         img = np.roll(img, y_roll, axis=-1)
 
-    # if train and (crop_size_x or crop_size_y):
-    #     img = img[:,:,rnd_x:rnd_x+crop_size_x, rnd_y:rnd_y+crop_size_y]
-
     if crop_size_x or crop_size_y:
```
Comment (Alexey-Kamenev) [inline modulus/experimental/resdiff/training/img_utils.py:112]: To me, removing `train` condition (as in the commented out code) is a bit strange - do they want to apply random crops during evaluation too?

**Review by Alexey-Kamenev — APPROVED** [review]
LGTM

File: modulus/experimental/resdiff/generate.py [hunk 4]
```diff
@@ -318,278 +315,154 @@ def load_pickle(network_pkl, rank):
...
+@hydra.main(version_base="1.2", config_path="conf", config_name="config_generate")
+def main(cfg: DictConfig) -> None:
+    """Generate random images using the techniques described in the paper
+    "Elucidating the Design Space of Diffusion-Based Generative Models".
+    """
 
-    opts = dnnlib.EasyDict(kwargs)
+    # Parse options
+    res_ckpt_filename = getattr(cfg, "res_ckpt_filename")
+    reg_ckpt_filename = getattr(cfg, "reg_ckpt_filename")
```
Comment (daviddpruitt) [inline modulus/experimental/resdiff/generate.py:388]: Since the checkpoint files are required shouldn't these be check rather than assign a default value?

Comment (mnabian) [reply modulus/experimental/resdiff/generate.py:388]: I'm not sure if I understand this correctly, but these are required args and don't have default values.

File: modulus/experimental/resdiff/training/img_utils.py (hunk 3 above)
Comment (mnabian) [reply modulus/experimental/resdiff/training/img_utils.py:112]: Yeah I'm not sure what the expected behavior is for the evaluation. I will add a TODO comment here to have an eye on it.

File: modulus/experimental/resdiff/conf/config_generate.yaml (hunk 1 above)
Comment (mnabian) [reply modulus/experimental/resdiff/conf/config_generate.yaml:43]: Sure, will remove this

**Comment by mnabian** [comment]
Hey @daviddpruitt @loliverhennigh, thanks for reviewing! Do you have any other comments?

**Pull request merged**